### PR TITLE
ozone: thumbnails support and various improvements

### PR DESCRIPTION
**Pull request by natinusala**
- The sidebar now collapses when entering a playlist
- A new setting was added to have the sidebar permanently collapsed
- A right sidebar now opens when entering a playlist, with a thumbnail and content metadata
- Content metadata can be replaced by the "left thumbnail"
- Corresponding settings have been renamed if the current menu driver is ozone
- Some various ticker and wrapping improvements were made

### CODE REVIEW

**Comment by natinusala** [comment]
Associated PR to retroarch-assets: https://github.com/libretro/retroarch-assets/pull/297

File: menu/drivers/ozone/ozone.c [hunk 1]
```diff
@@ -330,6 +374,254 @@ static void ozone_free(void *data)
...
+      if (!core_name || string_is_equal(core_name, "DETECT"))
+         core_label = msg_hash_to_str(MSG_AUTODETECT);
+      else
+         core_label = core_name;
+
+      snprintf(ozone->selection_core_name, sizeof(ozone->selection_core_name),
+         "%s", core_label);
+
+      /* Fill play time if applicable */
+      if (settings->bools.content_runtime_log)
```
Comment (jdgleaver) [inline menu/drivers/ozone/ozone.c]: This should be:

``` c
if (settings->bools.content_runtime_log || settings->bools.content_runtime_log_aggregate)
```

:slightly_smiling_face:

File: menu/menu_displaylist.c [hunk 2]
```diff
@@ -104,14 +104,6 @@ static enum msg_hash_enums new_type     = MSG_UNKNOWN;
...
 #define PL_LABEL_SPACER_RGUI    " | "
-#if defined(__APPLE__)
-/* UTF-8 support is currently broken on Apple devices... */
-#define PL_LABEL_SPACER_OZONE   "   |   "
-#else
-/* <EM SPACE><BULLET><EM SPACE>
- * UCN equivalent: "\u2003\u2022\u2003" */
-#define PL_LABEL_SPACER_OZONE   "\xE2\x80\x83\xE2\x80\xA2\xE2\x80\x83"
-#endif
 #define PL_LABEL_SPACER_MAXLEN 37
```
Comment (jdgleaver) [inline menu/menu_displaylist.c]: Could you reduce the size of `PL_LABEL_SPACER_MAXLEN` accordingly?

Comment (natinusala) [reply menu/menu_displaylist.c]: What would be the new size? 7 ?

**Comment by RobLoach** [comment]
Looks great! When the thumbnail is there, but the Thumbnail Setting is disabled, it displays "No thumbnail available". Could likely display nothing when an item is disabled.

thumb-disabled

**Comment by natinusala** [comment]
@RobLoach I did that because not displaying anything looked like wasted screen space, since the sidebar is here, might aswell display _something_. It also shows that thumbnails are a thing and would be placed here. Same for Play time and Last Played, the sidebar would look all empty and quite sad

**Comment by jdgleaver** [comment]
@natinusala Just a tiny heads-up: I'll be adding thumbnail support for mixed content playlists (history, favourties, etc.) to RGUI and XMB soon. This would be trivial to add here, to Ozone. You might want to do a quick copy/paste when I make the relevant PRs (I'll notify you when it's done, if you like)

**Comment by natinusala** [comment]
Sure, notify me and I will add it to ozone! I also need to implement savestate thumbnails which are missing, but it will be entirely different from what I did there (and quite fancy too)

File: menu/menu_displaylist.c (hunk 2 above)
Comment (jdgleaver) [reply menu/menu_displaylist.c]: Almost - you forgot the null terminator! Should be 8, I think.

(I mean, you could just leave it as-is, but might as well save a few bytes of RAM)

Comment (natinusala) [reply menu/menu_displaylist.c]: I was not sure of the meaning of that value because 37 seems oddly large for the ozone spacer. Fixing it now

**Comment by jdgleaver** [comment]
Okay, will do.

(This whole thing you've done is absolutely beautiful, by the way!)

File: menu/menu_displaylist.c (hunk 2 above)
Comment (jdgleaver) [reply menu/menu_displaylist.c]: Actually, you're right - I just realised that I set the size too large to begin with... :disappointed: 

But you've fixed it now, so it's all good!

**Comment by jdgleaver** [comment]
@natinusala Just a trivial bit of feedback: would it be possible to implement some kind of word wrapping for long core names? (Probably annoying to do, but everything else looks so nice)

Screenshot_2019-03-11_15-09-36

(Also, is the thumbnail slightly too small when running at lower display resolutions?)

**Comment by natinusala** [comment]
@jdgleaver you're right, I fixed it here : https://github.com/libretro/RetroArch/pull/8446
Thumbnails should be a little small for lower resolutions, I will fix it when implementing DPI scaling
Thanks for the feedback

**Comment by jdgleaver** [comment]
This is great - thank you so much!

**Pull request merged**